### PR TITLE
CORE-15087 update platform version 5.1

### PR DESCRIPTION
**Pull request by vlajos**
CORE-15087 update platform version to 50100 for 5.1 according spec. Centralize it. Make tests less-dependent of the main number.

### CODE REVIEW

File: libs/platform-info/src/main/kotlin/net/corda/libs/platform/impl/PlatformInfoProviderImpl.kt [hunk 1]
```diff
@@ -12,9 +12,9 @@ class PlatformInfoProviderImpl @Activate constructor(
 ) : PlatformInfoProvider {
 
     internal companion object {
-        const val STUB_PLATFORM_VERSION = 50000
+        const val STUB_PLATFORM_VERSION = 50100
 
-        private const val DEFAULT_PLATFORM_VERSION = 50000
+        private const val DEFAULT_PLATFORM_VERSION = 50100
```
Comment (vlajos) [inline libs/platform-info/src/main/kotlin/net/corda/libs/platform/impl/PlatformInfoProviderImpl.kt:17]: Are not these supposed to be derived somehow from the root gradle's platform version?

Comment (simon-johnson-r3) [reply libs/platform-info/src/main/kotlin/net/corda/libs/platform/impl/PlatformInfoProviderImpl.kt:17]: The long term plan for these is not yet determined. I think this is fine for now, the inconvenience means we don't forget about it. Although I appreciate that's easier to say if you're not the one having to do something about it...

Comment (vlajos) [reply libs/platform-info/src/main/kotlin/net/corda/libs/platform/impl/PlatformInfoProviderImpl.kt:17]: Thanks for explaining.

**Review by simon-johnson-r3 — APPROVED** [review]
Ideally we'd have the stable flag in place here too. But for now I think this is the best we can achieve.

**Review by thiagoviana — APPROVED** [review]
LGTM - Flow Worker Related

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Pull request merged**